### PR TITLE
extend button api and type attribute deprecation

### PR DESCRIPTION
**Pull request by nelix**
screen shot 2015-02-03 at 11 52 27 am

Basic api for the main types of buttons I saw in stack client, willing to take this further.
I don't like the way the colours work because the only overridable colours are listed by name, but thats an existing issue. I'm open for suggestions on a naming scheme that is easier to override for colours.

### CODE REVIEW

File: packages/button/button.scss [hunk 1]
```diff
@@ -1,7 +1,22 @@
+@import "colors";
+
 $button-font-family: 'Open Sans', 'Helvetica Neue', 'Arial', 'Helvetica', 'sans-serif' !default;
 $button-font-weight: 400 !default;
 $button-font-size: .8em !default;
 
+// TODO: these colors need work... reverse engineering them from stack/brief?
```
Comment (nelix) [inline packages/button/button.scss]: I don't like this

File: packages/button/example.jsx [hunk 2]
```diff
@@ -25,9 +25,15 @@ module.exports = React.createClass({
         <br/>
         Clicked {this.state.count} time{this.state.count === 1 ? '' : 's'}
         <br/><br/>
-        <Button type="huedemo">Has background-color support</Button>
+        <Button className="Button--huedemo">Has the ability to hack in varients</Button>
```
Comment (grapegravity) [inline packages/button/example.jsx]: The correct spelling here is `variants`. English is weird!

File: packages/button/button.scss [hunk 3]
```diff
@@ -34,4 +49,27 @@ $button-font-size: .8em !default;
   &:hover {
     opacity: .9;
   }
-}
\ No newline at end of file
+}
+
+.Button--success {
+  background: $color-button-success-background;
```
Comment (grapegravity) [inline packages/button/button.scss]: This needs to be `background-color` as using `background` removes the gradient overlay.

File: packages/button/button.scss (hunk 1 above)
Comment (grapegravity) [inline packages/button/button.scss]: I genuinely think we need to take a leaf from Bootstrap’s book and have colours named for purpose, so they can be overridden at a higher level.

File: packages/button/button.scss [hunk 4]
```diff
@@ -34,4 +47,27 @@ $button-font-size: .8em !default;
...
     opacity: .9;
   }
-}
\ No newline at end of file
+}
+
+.Button--success {
+  color: $color-button-success;
+  background-color: $color-button-success-background;
+  border-bottom-color: darken($color-button-success, 15%);
```
Comment (grapegravity) [inline packages/button/button.scss]: Why does this have a `border-bottom-color`?

Comment (nelix) [inline packages/button/button.scss]: the styles are taken direct from stack

File: packages/button/index.jsx [hunk 5]
```diff
@@ -2,19 +2,57 @@
...
+    small: React.PropTypes.bool,
+    skeleton: React.PropTypes.bool,
+    success: React.PropTypes.bool,
+    cancel: React.PropTypes.bool,
+    type: function(props, propName) {
+      if (
+        propName in props &&
+        !_.contains(validTypes, props[propName])
+      ) {
+        // type is depricated because it overlaps with the built in html attribute
```
Comment (grapegravity) [inline packages/button/index.jsx]: Would you mind correcting `depricated` to `deprecated` here?

**Comment by grapegravity** [comment]
I think this is pretty much ready to merge. Sorry for the on again-off again reviewing :stuck_out_tongue_winking_eye:

**Pull request merged**